packages/ckeditor5-utils/src/observablemixin.ts

```diff
@@ -906,3 +963,122 @@ function attachBindToListeners( observable, toBindings ) {
  * @method #decorate
  * @param {String} methodName Name of the method to decorate.
  */
+
+interface SingleBindChain<TKey extends string, TVal> {
+	toMany<O extends Observable, K extends keyof O>(
+		observables: readonly O[],
+		key: K,
+		callback: ( ...values: O[ K ][] ) => TVal
+	): void;
+
+	to<O extends Observable & { [ P in TKey ]: TVal }>(
```

Comment:
There can be many combinations here, like `x.bind( 'x' ).to( c1, c2, 'y', c3 )` which would equal to `x.bind( 'x' ).to( c1, 'x', c2, 'y', c3, 'x' )`. Unless we create a generator for these typings, let's keep only the common combinations up to 4 observables (the greatest number I've found in our code-base).